src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/AddCarry.PlatformNotSupported.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System;
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.Intrinsics.X86
+{
+    /// <summary>
+    /// This class provides access to Intel ADDCARRY hardware instructions via intrinsics
+    /// </summary>
+    [CLSCompliant(false)]
+    public abstract class AddCarry : Sse42
+    {
+        internal AddCarry() { }
+
+        public new static bool IsSupported { [Intrinsic] get { return false; } }
+
+        public new abstract class X64 : Sse41.X64
+        {
+            internal X64() { }
+
+            public new static bool IsSupported { [Intrinsic] get { return false; } }
+
+            /// <summary>
+            /// unsigned char _addcarry_u64 (unsigned char c_in, unsigned __int64 a, unsigned __int64 b, unsigned __int64 * out)
+            ///   ADC reg64, reg64
+            /// This intrinisc is only available on 64-bit processes
+            /// </summary>
+            public static unsafe byte AddWithCarry(byte carryIn, ulong left, ulong right, ulong* result) { throw new PlatformNotSupportedException(); }
```

Comment:
If you are looking to improve `BigInteger`, you should start with the `uint` overload. Switching to `ulong` will be a more involved change and make the perf comparison harder to make.

Reply:
Yes `uint` is likely the one I want. Will update later.

Reply:
Actually `ulong` would be more interesting in the case of `BigInteger`. The current generated code is already pretty reasonable and the only way `AddWithCarry` is likely to help is by switching that code to use `ulong` rather than `uint`, that's something that today cannot be done efficiently as it would require branching. Another option is to unroll the loop but that poses similar problems as using `ulong` so you may just do `ulong` from start.